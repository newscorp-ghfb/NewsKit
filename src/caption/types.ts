import {LogicalProps} from '../utils/logical-properties';
import {MQ} from '../utils/style';

// Caption component will be rebuilt in https://nidigitalsolutions.jira.com/browse/PPDSC-2091 to introduce breaking changes on the interface.
export interface CaptionProps {
  children: React.ReactNode;
  captionText?: string;
  creditText?: string;
  overrides?: {
    typographyPreset?: MQ<string>;
    stylePreset?: MQ<string>;
    spaceStack?: MQ<string>; // LOGICAL_PROPS_TO_DO: Used as the gap between items. Should be renamed.
    /**
     * @deprecated This property is deprecated and will be removed in the next major release. Use `paddingInline` & `paddingBlock` instead.
     */
    spaceInset?: MQ<string>;
    credit?: {
      typographyPreset?: MQ<string>;
      stylePreset?: MQ<string>;
    };
  } & LogicalProps;
}
