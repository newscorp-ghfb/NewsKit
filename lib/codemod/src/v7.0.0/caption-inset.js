const {replaceComponent} = require('../utils/component-utils');
const {updateOverrides} = require('../utils/update-spaceInset-props-overrides');
const {replaceSpaceInset} = require('../utils/replace-space-inset');

module.exports = function transformer(file, api) {
  const jscodeshift = api.jscodeshift;
  const root = jscodeshift(file.source);
  const source = root.toSource();

  updateOverrides(root, jscodeshift, source, 'CaptionInset', overrides => {
    // CaptionInset had this padding
    const defaultPadding = {
      paddingBlock: {
        xs: 'space040',
        md: 'space050',
      },
      paddingInline: {
        xs: 'space040',
        md: 'space050', 
      },
    };

    // when there is no overrides prop
    // we want to add some defaults
    if (!overrides) {
      return defaultPadding;
    }

    const {spaceInset} = overrides;

    // does not have spaceInset so we add it manually
    if (!spaceInset) {
      return {
        ...overrides,
        ...defaultPadding,
      };
    }

    return replaceSpaceInset(overrides);
  });

  replaceComponent(jscodeshift, root, 'CaptionInset', 'Caption');

  return root.toSource();
};
