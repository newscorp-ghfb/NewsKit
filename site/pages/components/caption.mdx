import Layout from '../../components/layout';
import Playground from '../../components/playground';
import {Prop} from '../../components/prop';
import {TextBoxSwatch} from '../../components/theming-values/colors';
import {LegacyBlock} from '../../components/legacy-block';
import {CommonLogicalPropsMDX} from '../../components/component-api/common-logical-props';

import {Caption, CaptionInset} from 'newskit';

export default Layout;

# Caption

## Overview

A caption is a sentence often added to an image or video to describe or explain what the image or video is showing. Sometimes a credit to the owner of the source is added. The caption has two main elements, the caption text and an optional credit text.

## Usage

The `Caption` component is generally placed below the related media on the page. The component takes text as its child.

## Variants

The are two variants of the caption component - a caption without inset `<Caption/>` and a caption with inset `<CaptionInset/>`.

## Props

<LegacyBlock marginTop="sizing040">
  <Prop name="children" type="object" required>
    This is the text of the Caption.
  </Prop>
  <Prop name="creditText" type="string">
    If provided, supplies the content of the credit portion of the component.
  </Prop>
  <Prop name="overrides" type="object">
    If provided, overrides the respective presets for the component and provided
    elements.
      <Prop name="stylePreset" type="MQ<string>">
        If provided, this overrides the style preset of the component.
      </Prop>
      <Prop name="typographyPreset" type="MQ<string>">
        If provided, this overrides the typography preset of the component..
      </Prop>
      <Prop name="spaceStack" type="MQ<string>">
        If provided, this overrides the spacing between the caption and the credit.
      </Prop>
      <Prop name="credit" type="object" >
        If provided, this overrides the styling of the credit portion of the component.
          <Prop name="typographyPreset" type="MQ<string>">
            If provided, this overrides the typography preset of the credit text.
          </Prop>
          <Prop name="stylePreset" type="MQ<string>">
            If provided, this overrides the style preset of the credit text.
          </Prop>
      </Prop>
      <Prop name="spaceInset" type="MQ<string>">
       This property is deprecated. Use paddingInline and paddingBlock instead. If provided, this overrides the padding of the Caption. This override should only be used with CaptionInset. Otherwise, spaceInset040 or spaceInset050  is used depending on the `size` prop.
      </Prop>
      <CommonLogicalPropsMDX/>

  </Prop>
</LegacyBlock>

## Playground

<Playground
  componentName={['Caption', 'CaptionInset']}
  newskitPath="/caption"
  component={[
    Caption,
    props => (
      <div style={{border: '1px solid black;'}}>
        <CaptionInset {...props} />
      </div>
    ),
  ]}
  knobs={[
    {
      name: 'Caption content',
      propName: 'children',
      value: 'This is an example of a caption',
    },
    {
      name: 'Credit content',
      propName: 'creditText',
      value: 'This is an example of a credit',
    },
    {
      name: 'Caption style preset override',
      propName: 'overrides',
      options: [
        {
          label: 'Default',
          value: {
            stylePreset: undefined,
          },
          isDefault: true,
        },
        {
          label: 'inkBrand010',
          value: {
            stylePreset: 'inkBrand010',
          },
        },
      ],
    },
    {
      name: 'Caption typography preset override',
      propName: 'overrides',
      options: [
        {
          label: 'Default',
          value: {
            typographyPreset: undefined,
          },
          isDefault: true,
        },
        {
          label: 'meta030',
          value: {
            typographyPreset: 'meta030',
          },
        },
      ],
    },
    {
      name: 'Caption spaceStack override',
      propName: 'overrides',
      options: [
        {
          label: 'Default',
          value: {
            spaceStack: undefined,
          },
          isDefault: true,
        },
        {
          label: 'space090',
          value: {
            spaceStack: 'space090',
          },
        },
      ],
    },
    {
      name: 'Credit style preset override',
      propName: 'overrides',
      options: [
        {
          label: 'Default',
          value: {
            credit: {
              stylePreset: undefined,
            },
          },
          isDefault: true,
        },
        {
          label: 'inkContrast',
          value: {
            credit: {
              stylePreset: 'inkContrast',
            },
          },
        },
      ],
    },
    {
      name: 'Credit typography preset override',
      propName: 'overrides',
      options: [
        {
          label: 'Default',
          value: {
            credit: {
              typographyPreset: undefined,
            },
          },
          isDefault: true,
        },
        {
          label: 'utilityMeta020',
          value: {
            credit: {
              typographyPreset: 'utilityMeta020',
            },
          },
        },
      ],
    },
  ]}
/>
