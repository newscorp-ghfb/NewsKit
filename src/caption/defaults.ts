export default {
  caption: {
    typographyPreset: 'editorialCaption010',
    stylePreset: 'inkBase',
    spaceStack: 'space040',
    credit: {
      typographyPreset: 'utilityMeta010',
      stylePreset: 'uppercaseInkSubtle',
    },
  },
};
