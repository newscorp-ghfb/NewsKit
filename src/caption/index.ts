export * from './caption';
export * from './types';
