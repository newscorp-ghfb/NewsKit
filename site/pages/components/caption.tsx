import React from 'react';
import {Caption} from 'newskit';
import {MetaStatus} from '../../components/meta/types';
import {LayoutProps} from '../../components/layout';
import {ComponentPageTemplate} from '../../templates/component-page-template/component-page-template';
import {getIllustrationComponent} from '../../components/illustrations/illustration-loader';
import {OverridesRowsProps} from '../../components/component-api';
import {commonLogicalProps} from '../../components/component-api/common-logical-props';
import {UsageKind} from '../../components/usage-card';

const CaptionComponent = (layoutProps: LayoutProps) => (
  <ComponentPageTemplate
    headTags={{
      title: 'Caption',
      description:
        'A caption is a text description of an image or video to describe what is showing.',
    }}
    layoutProps={layoutProps}
    pageIntroduction={{
      type: 'Text',
      name: 'Caption',
      hero: {
        illustration: 'components/caption-illustration',
      },
      introduction: `A caption is a text description of an image or video to describe what is showing.`,
    }}
    componentDefaultsKey="Caption"
    meta={{
      status: MetaStatus.Supported,
      introduced: 'v0.19.0',
      codeUrl: 'https://github.com/newscorp-ghfb/newskit/tree/main/src/caption',
      storybookId: 'components-caption--story-caption-default',
      figmaUrl:
        'https://www.figma.com/file/FSbCQa6SzVR3K48ZWLeD77/%F0%9F%9F%A2-NK-Web-Components?node-id=1967%3A5713&t=4rcwseDIwrZdWokj-0',
    }}
    interactiveDemo={{
      introduction:
        'This demo lets you preview the caption component, its variations, and configuration options.',
      playground: {
        componentName: 'Caption',
        component: props => <Caption {...props} />,
        knobs: [
          {
            name: 'Caption text',
            propName: 'children',
            value: 'Hathersage Moor in the Peak District',
          },
          {
            name: 'Credit text',
            propName: 'creditText',
            value: 'Credit by Matthew Taylor/Alamy',
          },
        ],
      },
    }}
    anatomy={{
      introduction:
        'The caption contains one required element and one optional element.',
      rows: [
        {
          name: 'Caption text',
          description: 'The text of the caption describing the image or video',
          component: 'Text Block',
          optional: undefined,
        },
        {
          name: 'Credit text',
          description: 'Optional text, crediting the source',
          component: 'Text Block',
          optional: true,
        },
      ],
      media: getIllustrationComponent('components/caption/anatomy/anatomy'),
    }}
    options={{
      introduction:
        'The caption has options that can be used to provide an appropriate experience for different use cases.',
      cards: [
        {
          title: 'Credit text',
          description:
            'A credit to the source can be added underneath the caption text.',
          media: getIllustrationComponent(
            'components/caption/options/credit-text',
          ),
        },
        {
          title: 'Inset spacing',
          description:
            'The caption can be inset with increased spacing. There are two sizes of inset: XS and MD.',
          media: getIllustrationComponent(
            'components/caption/options/inset-spacing',
          ),
        },
      ],
    }}
    usage={{
      introduction:
        'The following guidance describes how and when to use the caption component appropriately.',
      cards: [
        {
          title: 'Donâ€™t include captions inside images',
          description:
            'Legibility can be poor and screen readers cannot read the words. Use live text instead.',
          kind: UsageKind.DONT,
          media: getIllustrationComponent(
            'components/caption/usage/inside-images',
          ),
        },
      ],
    }}
    accessibility={{
      introduction:
        'Captions help ensure that media like images and video are accessible to everyone by providing supplemental information about what the image is conveying. Captions are displayed within the main content and can be read by assistive technology.',
    }}
    componentAPI={{
      components: [
        {
          title: 'Caption',
          propsSummary:
            'The caption has a range of props that can be used to define an appropriate experience for different use cases.',
          overridesSummary:
            'The caption has a range of predefined elements and attributes that can be overridden to define its appearance.',
          propsRows: [
            {
              name: 'children',
              type: 'ReactNode',
              required: true,
              description: 'The text of the caption',
            },
            {
              name: 'creditText',
              type: 'string',
              description:
                'If provided, supplies the content for the component credit',
            },
          ],
          overridesRows: [
            {
              attribute: 'stylePreset',
              type: 'MQ<string>',
              default: 'inkBase',
              description:
                'If provided, this overrides the style preset of the caption',
            },
            {
              attribute: 'typographyPreset',
              type: 'MQ<string>',
              default: 'editorialCaption010',
              description:
                'If provided, this overrides the typography preset of the caption',
            },
            {
              attribute: 'spaceStack',
              type: 'MQ<string>',
              default: 'space040',
              description:
                'If provided, this overrides the spacing between the caption and the credit',
            },
            {
              attribute: 'credit.stylePreset',
              type: 'MQ<string>',
              default: 'uppercaseInkSubtle',
              description:
                'If provided, this overrides the style preset of the credit text',
            },
            {
              attribute: 'credit.typographyPreset',
              type: 'MQ<string>',
              default: 'utilityMeta010',
              description:
                'If provided, this overrides the typography preset of the credit text',
            },
            ...(commonLogicalProps() as OverridesRowsProps[]),
          ],
        },
        {
          title: 'CaptionInset',
          summary:
            'The captionInset has a range of predefined elements and attributes that can be overridden to define its appearance.',
          overridesRows: [
            {
              attribute: 'stylePreset',
              type: 'MQ<string>',
              default: 'inkBase',
              description:
                'If provided, this overrides the style preset of the captionInset',
            },
            {
              attribute: 'typographyPreset',
              type: 'MQ<string>',
              default: 'editorialCaption010',
              description:
                'If provided, this overrides the typography preset of the captionInset',
            },
            {
              attribute: 'spaceStack',
              type: 'MQ<string>',
              default: 'space040',
              description:
                'If provided, this overrides the spacing between the captionInset and the credit',
            },
            {
              attribute: 'spaceInset (deprecated)',
              type: 'MQ<string>',
              default: '{xs: spaceInset040, md: spaceInset050}',
              description:
                'If provided, this overrides the padding of the captionInset.',
            },
          ],
        },
      ],
    }}
    related={{
      related: ['Text Block'],
    }}
  />
);

export default CaptionComponent;
