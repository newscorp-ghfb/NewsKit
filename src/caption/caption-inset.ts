import {withDefaultProps} from '../utils/with-default-props';
import {withOwnTheme} from '../utils/with-own-theme';
import {Caption} from './caption';
import defaults from './defaults';

export const CaptionInset = withOwnTheme(
  withDefaultProps(Caption, {}, 'captionInset'),
)({
  defaults,
});
