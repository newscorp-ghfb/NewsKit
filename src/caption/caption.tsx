import React from 'react';
import {CaptionProps} from './types';
import {Block} from '../block';
import {getToken} from '../utils/get-token';
import {useTheme} from '../theme';
import {TextBlock} from '../text-block';
import {withOwnTheme} from '../utils/with-own-theme';
import defaults from './defaults';
import {extractLogicalPropsFromOverrides} from '../utils/logical-properties';

const ThemelessCaption = React.forwardRef<HTMLDivElement, CaptionProps>(
  ({overrides, children, creditText, ...rest}, ref) => {
    const theme = useTheme();
    const captionGap =
      creditText && getToken({theme, overrides}, 'caption', '', 'spaceStack');

    const captionStylePreset = getToken(
      {theme, overrides},
      'caption',
      '',
      'stylePreset',
    );

    const captionTypographyPreset = getToken(
      {theme, overrides},
      'caption',
      '',
      'typographyPreset',
    );

    const creditTypographyPreset = getToken(
      {theme, overrides},
      'caption.credit',
      'credit',
      'typographyPreset',
    );
    const creditStylePreset = getToken(
      {theme, overrides},
      'caption.credit',
      'credit',
      'stylePreset',
    );

    const captionInset = getToken(
      {theme, overrides},
      'caption',
      '',
      'spaceInset',
    );

    const logicalProps = extractLogicalPropsFromOverrides(overrides);

    return (
      <Block spaceInset={captionInset} ref={ref} {...logicalProps} {...rest}>
        <TextBlock
          stylePreset={captionStylePreset}
          typographyPreset={captionTypographyPreset}
          marginBlockEnd={captionGap}
        >
          {children}
        </TextBlock>
        {creditText && (
          <TextBlock
            stylePreset={creditStylePreset}
            typographyPreset={creditTypographyPreset}
          >
            {creditText}
          </TextBlock>
        )}
      </Block>
    );
  },
);

// Caption will be rebuilt in https://nidigitalsolutions.jira.com/browse/PPDSC-2091
export const Caption = withOwnTheme(ThemelessCaption)({
  defaults,
});
ThemelessCaption.displayName = 'Caption';
