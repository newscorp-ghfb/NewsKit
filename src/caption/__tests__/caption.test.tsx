import {Caption} from '../caption';
import {CaptionInset} from '../caption-inset';
import {renderToFragmentWithTheme} from '../../test/test-utils';

describe('Caption', () => {
  test('renders correctly with default props and caption and credit text', () => {
    const fragment = renderToFragmentWithTheme(Caption, {
      children: 'test',
      creditText: 'this is credit text',
    });
    expect(fragment).toMatchSnapshot();
  });
});

test('renders correctly with default props and only caption text', () => {
  const fragment = renderToFragmentWithTheme(Caption, {
    children: 'test',
  });

  expect(fragment).toMatchSnapshot();
});

test('renders with additional HTML attributes', () => {
  const fragment = renderToFragmentWithTheme(Caption, {
    id: 'caption-id',
  });

  expect(fragment).toMatchSnapshot();
});

test('renders correctly with overriden props and caption and credit text', () => {
  const fragment = renderToFragmentWithTheme(Caption, {
    children: 'test',
    creditText: 'this is credit text',
    overrides: {
      typographyPreset: 'utilitySubheading050',
      stylePreset: 'inkContrast',
      spaceStack: 'space090',
      credit: {
        typographyPreset: 'utilityMeta010',
        stylePreset: 'uppercaseInkContrast',
      },
    },
  });
  expect(fragment).toMatchSnapshot();
});

describe('CaptionInset', () => {
  test('renders correctly with default props and caption and credit text', () => {
    const fragment = renderToFragmentWithTheme(CaptionInset, {
      children: 'test',
      creditText: 'this is credit text',
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders correctly with default props and only caption text', () => {
    const fragment = renderToFragmentWithTheme(CaptionInset, {
      children: 'test',
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders correctly with overriden props and caption and credit text', () => {
    const fragment = renderToFragmentWithTheme(CaptionInset, {
      children: 'test',
      creditText: 'this is credit text',
      overrides: {
        typographyPreset: 'utilitySubheading050',
        stylePreset: 'inkContrast',
        spaceStack: 'space090',
        credit: {
          typographyPreset: 'utilityMeta010',
          stylePreset: 'uppercaseInkContrast',
        },
      },
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders correctly with logical props overrides ', () => {
    const fragment = renderToFragmentWithTheme(CaptionInset, {
      children: 'test',
      creditText: 'this is credit text',
      overrides: {
        marginBlock: '30px',
        paddingBlock: '30px',
      },
    });
    expect(fragment).toMatchSnapshot();
  });
});
