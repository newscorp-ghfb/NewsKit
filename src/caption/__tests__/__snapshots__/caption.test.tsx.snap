// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caption renders correctly with default props and caption and credit text 1`] = `
<DocumentFragment>
  .emotion-1 {
  margin: 0;
  color: #3B3B3B;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
  margin-block-end: 16px;
}

.emotion-1 svg {
  fill: #3B3B3B;
}

.emotion-1::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-1::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

.emotion-2 {
  margin: 0;
  color: #6A6A6A;
  text-transform: uppercase;
  font-family: "DM Sans",sans-serif;
  font-size: 12px;
  line-height: 18px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
}

.emotion-2 svg {
  fill: #6A6A6A;
}

.emotion-2::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-2::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

<div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      test
    </p>
    <p
      class="emotion-2"
    >
      this is credit text
    </p>
  </div>
</DocumentFragment>
`;

exports[`CaptionInset renders correctly with default props and caption and credit text 1`] = `
<DocumentFragment>
  @media screen and (max-width: 767px) {
  .emotion-0 {
    padding: 16px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-0 {
    padding: 24px;
  }
}

.emotion-1 {
  margin: 0;
  color: #3B3B3B;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
  margin-block-end: 16px;
}

.emotion-1 svg {
  fill: #3B3B3B;
}

.emotion-1::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-1::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

.emotion-2 {
  margin: 0;
  color: #6A6A6A;
  text-transform: uppercase;
  font-family: "DM Sans",sans-serif;
  font-size: 12px;
  line-height: 18px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
}

.emotion-2 svg {
  fill: #6A6A6A;
}

.emotion-2::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-2::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

<div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      test
    </p>
    <p
      class="emotion-2"
    >
      this is credit text
    </p>
  </div>
</DocumentFragment>
`;

exports[`CaptionInset renders correctly with default props and only caption text 1`] = `
<DocumentFragment>
  @media screen and (max-width: 767px) {
  .emotion-0 {
    padding: 16px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-0 {
    padding: 24px;
  }
}

.emotion-1 {
  margin: 0;
  color: #3B3B3B;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
}

.emotion-1 svg {
  fill: #3B3B3B;
}

.emotion-1::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-1::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

<div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      test
    </p>
  </div>
</DocumentFragment>
`;

exports[`CaptionInset renders correctly with logical props overrides  1`] = `
<DocumentFragment>
  .emotion-0 {
  margin-block: 30px;
  padding-block: 30px;
}

@media screen and (max-width: 767px) {
  .emotion-0 {
    padding: 16px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-0 {
    padding: 24px;
  }
}

.emotion-1 {
  margin: 0;
  color: #3B3B3B;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
  margin-block-end: 16px;
}

.emotion-1 svg {
  fill: #3B3B3B;
}

.emotion-1::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-1::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

.emotion-2 {
  margin: 0;
  color: #6A6A6A;
  text-transform: uppercase;
  font-family: "DM Sans",sans-serif;
  font-size: 12px;
  line-height: 18px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
}

.emotion-2 svg {
  fill: #6A6A6A;
}

.emotion-2::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-2::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

<div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      test
    </p>
    <p
      class="emotion-2"
    >
      this is credit text
    </p>
  </div>
</DocumentFragment>
`;

exports[`CaptionInset renders correctly with overriden props and caption and credit text 1`] = `
<DocumentFragment>
  @media screen and (max-width: 767px) {
  .emotion-0 {
    padding: 16px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-0 {
    padding: 24px;
  }
}

.emotion-1 {
  margin: 0;
  color: #111111;
  font-family: "Poppins",sans-serif;
  font-size: 22px;
  line-height: 24.75px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
  margin-block-end: 64px;
}

.emotion-1 svg {
  fill: #111111;
}

.emotion-1::before {
  content: '';
  margin-bottom: -0.2155em;
  display: block;
}

.emotion-1::after {
  content: '';
  margin-top: -0.2125em;
  display: block;
}

.emotion-2 {
  margin: 0;
  color: #111111;
  text-transform: uppercase;
  font-family: "DM Sans",sans-serif;
  font-size: 12px;
  line-height: 18px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
}

.emotion-2 svg {
  fill: #111111;
}

.emotion-2::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-2::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

<div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      test
    </p>
    <p
      class="emotion-2"
    >
      this is credit text
    </p>
  </div>
</DocumentFragment>
`;

exports[`renders correctly with default props and only caption text 1`] = `
<DocumentFragment>
  .emotion-1 {
  margin: 0;
  color: #3B3B3B;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
}

.emotion-1 svg {
  fill: #3B3B3B;
}

.emotion-1::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-1::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

<div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      test
    </p>
  </div>
</DocumentFragment>
`;

exports[`renders correctly with overriden props and caption and credit text 1`] = `
<DocumentFragment>
  .emotion-1 {
  margin: 0;
  color: #111111;
  font-family: "Poppins",sans-serif;
  font-size: 22px;
  line-height: 24.75px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
  margin-block-end: 64px;
}

.emotion-1 svg {
  fill: #111111;
}

.emotion-1::before {
  content: '';
  margin-bottom: -0.2155em;
  display: block;
}

.emotion-1::after {
  content: '';
  margin-top: -0.2125em;
  display: block;
}

.emotion-2 {
  margin: 0;
  color: #111111;
  text-transform: uppercase;
  font-family: "DM Sans",sans-serif;
  font-size: 12px;
  line-height: 18px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
}

.emotion-2 svg {
  fill: #111111;
}

.emotion-2::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-2::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

<div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      test
    </p>
    <p
      class="emotion-2"
    >
      this is credit text
    </p>
  </div>
</DocumentFragment>
`;

exports[`renders with additional HTML attributes 1`] = `
<DocumentFragment>
  .emotion-1 {
  margin: 0;
  color: #3B3B3B;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
}

.emotion-1 svg {
  fill: #3B3B3B;
}

.emotion-1::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-1::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

<div
    class="emotion-0"
    id="caption-id"
  >
    <p
      class="emotion-1"
    />
  </div>
</DocumentFragment>
`;
